cmd/server/decoder_test.go

```
@@ -0,0 +1,60 @@
+// TestRequestDecoderRawBody covers the transport, which the service-level
+// tests bypass entirely.
+//
+// A goa `Bytes` payload is decoded with encoding/json, and that accepts only a
+// base64 string for a byte slice — so the default decoder rejects an ordinary
+// JSON body before any handler runs. On this endpoint that meant every real
+// Auth0 delivery would have been refused, and the 400 would have told Auth0 to
+// stop retrying.
+func TestRequestDecoderRawBody(t *testing.T) {
+	t.Run("a JSON object body arrives verbatim", func(t *testing.T) {
+		payload := `{"id":"evt_1","type":"user.updated","data":{"object":{"user_id":"auth0|1"}}}`
+		request := httptest.NewRequest(http.MethodPost, "/webhooks/auth0/cdp-provisioning", strings.NewReader(payload))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)